cpp/cmake_modules/Finduriparser.cmake

```diff
@@ -0,0 +1,53 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+pkg_check_modules(uriparser_PC liburiparser)
+if(uriparser_PC_FOUND)
+  set(uriparser_INCLUDE_DIR "${uriparser_PC_INCLUDEDIR}")
+
+  list(APPEND uriparser_PC_LIBRARY_DIRS "${uriparser_PC_LIBDIR}")
+  find_library(uriparser_LIB uriparser
+               PATHS ${uriparser_PC_LIBRARY_DIRS}
+               NO_DEFAULT_PATH
+               PATH_SUFFIXES "${CMAKE_LIBRARY_ARCHITECTURE}")
+elseif(uriparser_ROOT)
+  message(STATUS "Using uriparser_ROOT: ${uriparser_ROOT}")
+  find_library(uriparser_LIB
+               NAMES uriparser
+               PATHS ${uriparser_ROOT} "${uriparser_ROOT}/Library"
```

Comment:
Why `Library`?

Reply:
Oh whoops, I copied this from another Find*.cmake, I'll remove that. Thanks for catching that.

Reply:
I see this in `FindBrotli.cmake` as well.
@xhochy, any reason to use `Library` specifically and not in `LIB_PATH_SUFFIXES`?

Reply:
To be clear, LIB_PATH_SUFFIXES is introduced here - I can just move "Library" there to cover these cases.

Reply:
Can you also update this in the following?

> FindGLOG.cmake,  Findc-aresAlt.cmake, FindZSTD.cmake, FindRE2.cmake, FindBrotli.cmake, FindgRPCAlt, FindLz4.cmake

Reply:
Sure, done.

Reply:
Ah, so the reason appears to be the windows build - you can see in the logs that things are installed to "<root>/Library/lib" (https://ci.appveyor.com/project/ApacheSoftwareFoundation/arrow/builds/23542439/job/35vae9ce9os74cie)

I'll revert that then.

Reply:
Can't you add `Library/lib` to the suffix?

Reply:
It looks like we need at least `Library/include` and `Library/bin` as well, then.

Reply:
I've made that change instead.

Reply:
Makes sense to add them as well. This is only needed for Windows.